kpet/schema.py

```diff
@@ -396,13 +396,20 @@ class List(Node):
     """
     List schema, with every element matching a single specified schema.
     """
-    def __init__(self, element_schema):
+    def __init__(self, element_schema, min_len=None):
         assert isinstance(element_schema, Node)
         super().__init__(list)
         self.element_schema = element_schema
+        self.min_len = min_len
 
     def validate(self, data):
         super().validate(data)
+
+        if self.min_len and len(data) < self.min_len:
```

Comment:
If you assume (and assert) that min_len is an integer, you would only need to check for `len(data) < self.min_len`.